File: app/controllers/pdf_reports_controller.rb
class PdfReportsController < ApplicationController
  def show
    # 1. For /pdf?type=biologics&demo=1, render a real PDF
    if params[:type] == "biologics" && params[:demo] == "1"
      require 'prawn'

      pdf = Prawn::Document.new
      pdf.text "Thomas IT - CoC / Biologics Demo PDF", size: 18, style: :bold
      pdf.text "Type: #{params[:type] || 'unknown'}"
      pdf.text "Demo: #{params[:demo] || 'N/A'}"
      pdf.text "SHA256: #{Digest::SHA256.hexdigest(Time.now.to_s)}"
      pdf.text "Generated: #{Time.now.utc.iso8601}"

      send_data pdf.render,
                filename: "biologics-demo.pdf",
                type: "application/pdf",
                disposition: "inline"
    else
      # 2. Fallback for plain /pdf request (still 200)
      render plain: "CoC / biologics PDF endpoint (Phase 11)", status: 200
    end
  end
end

File: app/controllers/pdf_reports_controller.r
# app/controllers/pdf_reports_controller.rb
class PdfReportsController < ApplicationController
  # Auth is optional; you can skip if you want this public
  # skip_before_action :authenticate_user!, only: [:show]

  def show
    respond_to do |format|
      format.html { render plain: "biologics / pdf stub" }
      format.pdf do
        require 'prawn'
        pdf = Prawn::Document.new
        pdf.text "THOMAS IT PHARMA TRANSPORT - PDF SERVICE (demo)"
        pdf.text "Type: #{params[:type]}"
        pdf.text "Demo: #{params[:demo]}"

        send_data pdf.render,
          filename: "pharma-demo-#{params[:type] || 'generic'}.pdf",
          type: 'application/pdf',
          disposition: 'inline'
      end
      # Important: explicitly render nothing for other formats so response is not empty
      format.* { head :not_acceptable }
    end
  end
end
